### PR TITLE
add comment functionality to portfolio

### PR DESCRIPTION
**Pull request by ThomasQuesadilla**
Users can add comments, limit number of comments shown, and delete comments from database. Comments are saved to DataStore.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -14,24 +14,90 @@
...
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import com.google.gson.Gson;
 import java.util.ArrayList;
 import java.util.Arrays;
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
```
Comment (SylviaZiyuZhang) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:32]: It doesn't particularly matter for this project, but keeping comments updated is usually helpful =)

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -12,13 +12,31 @@
...
+  messageElement.innerHTML = comment.message;
+
+  commentElement.appendChild(messageElement);
+  return commentElement
+}
+
+function deleteComments() {
+  const request = new Request('/delete-data', {method:'POST'})
+  fetch(request).then(getComments(0));
+}
```
Comment (SylviaZiyuZhang) [inline portfolio/src/main/webapp/script.js:42]: Looks good overall!

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -9,7 +9,7 @@
 <link rel="stylesheet" href="style.css">
 <script src="script.js"></script>
 
-<body onload="getComments()">
+<body onload="getComments(5)">
```
[the remark this answers is not in the document]
Comment (ThomasQuesadilla) [reply portfolio/src/main/webapp/index.html:12]: ooo thats super smart alright i'll try that out

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -14,24 +14,90 @@
...
-  ArrayList<String> comments = new ArrayList<>(Arrays.asList("Yo this website is soooooooo cool", "That's cap bro this website is trash", "nah bro youre trash")); 
-  
-  @Override public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException { 
+  @Override 
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException { 
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    maxComments = getMaxCommentChoice(request);
+    
+    ArrayList<Comment> comments = new ArrayList<>();
```
Comment (linguistrix) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:50]: If you know the number of elements beforehand, you can pass it to the constructor so it allocates that many.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -14,24 +14,90 @@
...
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    maxComments = DEFAULT_NUMBER_OF_COMMENTS;
+  }
   
-  ArrayList<String> comments = new ArrayList<>(Arrays.asList("Yo this website is soooooooo cool", "That's cap bro this website is trash", "nah bro youre trash")); 
-  
-  @Override public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException { 
+  @Override 
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException { 
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
```
Comment (linguistrix) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:46]: Can you not provide the max number of rows you want?

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -12,13 +12,31 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function getComments() {
-  fetch('/data').then(response => response.json()).then(comments => {
+/*Pulls comments and adds html/styling elements to display*/
```
Comment (linguistrix) [inline portfolio/src/main/webapp/script.js:15]: /*[space] ... [space]*/

**Pull request merged**